src/apps/watchfaces/OswAppWatchfaceMonotimer.cpp

```diff
@@ -0,0 +1,156 @@
+
+#include "./apps/watchfaces/OswAppWatchfaceMonotimer.h"
+// #define GIF_BG
+
+#include <gfx_util.h>
+#include <osw_app.h>
+#include <osw_config.h>
+#include <osw_config_keys.h>
+#include <osw_hal.h>
+
+#ifdef GIF_BG
+#include "./apps/_experiments/gif_player.h"
+#endif
+
+/**
+ * @brief Draw N shifted ticks around the clock.
+ *
+ * @param cx center x axis
+ * @param cy center y axis
+ * @param r1 radius from the begin of the tick.
+ * @param r2 radius from the end of the tick.
+ * @param nTicks number of ticks to draw
+ * @param shift amount of degrees the ticks are shifted clockwise from the top (0°)
+ * @param color color code
+ */
+void OswAppWatchfaceMonotimer::drawNShiftedTicks(Graphics2D *gfx, uint8_t cx, uint8_t cy, uint8_t r1, uint8_t r2, uint8_t nTicks, float shift, uint16_t color) {
+    float deltaAngle = 360.0 / nTicks;
+    for (uint16_t i=0; i<nTicks; ++i) {
+        gfx->drawTick(cx, cy, r1, r2, (i * deltaAngle) + shift, color);
+    }
+}
+
+/**
+ * @brief Draw N shifted ticks around the clock; skip every m-th.
+ *
+ * @param cx center x axis
+ * @param cy center y axis
+ * @param r1 radius from the begin of the tick.
+ * @param r2 radius from the end of the tick.
+ * @param nTicks number of ticks to draw
+ * @param shift amount of degrees the ticks are shifted clockwise from the top (0°)
+ * @param m do not draw every m-th tick
+ * @param color color code
+ */
+void OswAppWatchfaceMonotimer::drawNShiftedMaskedTicks(Graphics2D *gfx, uint8_t cx, uint8_t cy, uint8_t r1, uint8_t r2, uint8_t nTicks, float shift, uint16_t m, uint16_t color) {
+    float deltaAngle = 360.0 / nTicks;
+    for (uint16_t i=0; i<nTicks; ++i) {
+        if (i % m) {
+            gfx->drawTick(cx, cy, r1, r2, (i * deltaAngle) + shift, color);
+        }
+    }
+}
+
+void OswAppWatchfaceMonotimer::drawWatch() {
+    OswHal* hal = OswHal::getInstance();
+
+    // hours
+    hal->gfx()->drawNTicks(120, 120, 117, 100, 12, ui->getForegroundColor());
+    // 30 minutes
+    drawNShiftedTicks(hal->gfx(), 120, 120, 117, 105, 12, 360.0/24.0, ui->getForegroundColor());
+    // 15 minutes
+    drawNShiftedTicks(hal->gfx(), 120, 120, 110, 105, 24, 360.0/48.0, ui->getForegroundColor());
+    // 5 minutes
+    drawNShiftedMaskedTicks(hal->gfx(), 120, 120, 109, 108, 144, 0.0, 3, ui->getForegroundColor());
+
+    // hour labels
+    hal->gfx()->setTextSize(2);
+    hal->gfx()->setTextMiddleAligned();
+
+    static uint8_t positions[]= {
+        155, 45, 	// 01
+        180, 75, 	// 02
+        200, 120,	// 03
+        183, 165, 	// 04
+        155, 197, 	// 05
+        110, 210, 	// 06
+        65, 197, 	// 07
+        37, 165, 	// 08
+        23, 120, 	// 09
+        37, 75, 	// 10
+        65, 45, 	// 11
+        110, 30 	// 12
+    };
+
+    for (uint8_t i=0; i<12; ++i) {
+        hal->gfx()->setTextCursor(positions[i<<1], positions[(i<<1)|1]);
```

Comment:
What are you bit-shifting there? What is this code doing? :confused:

Reply:
Holy sh*t! Thats genius! Lol.

Reply:
Good - code !